resin_release_tool/balena_backend.py

```diff
@@ -0,0 +1,52 @@
+from typing import Dict
+from balena import exceptions
+
+
+class BalenaBackend:
+    """Wrapper around the balena_sdk to make it easier to swap it out for unit testing.
+
+    The SUPPORTED_API_VERSION is used to pin the expected Balena Api version used by the balena-sdk installation.
+    It is used to show a warning if the Api version changes, as this might introduce breaking changes.
+    """
+
+    SUPPORTED_API_VERSION = "v6"
+
+    def __init__(self, models, settings):
+        self.models = models
+        self.settings = settings
+
+    def get_application_info(self, id: int) -> Dict:
+        """https://www.balena.io/docs/reference/sdk/python-sdk/#function-get_by_idapp_id
+        note that the Example response documented there differs from the actual response
+        (which is missing device_type and commit)
+        """
+        return self.models.application.get_by_id(id)
+
+    def _get_application_base_request(self):
+        return self.models.application.base_request
+
+    def get_release_by_id(self, id: int) -> Dict:
+        return self._get_resource_by_id("release", id)
+
+    def get_device_type_by_id(self, id: int) -> Dict:
+        return self._get_resource_by_id("device_type", id)
+
+    def _get_resource_by_id(self, resource: str, id: int) -> Dict:
+        base_request = self._get_application_base_request()
+        params = {"filters": {"id": id}}
+        response = base_request.request(
+            resource, "GET", params=params, endpoint=self.settings.get("pine_endpoint")
+        )
+        if response["d"]:
+            assert (
+                len(response["d"]) == 1
+            ), f"getting device type by id should return one type {response[d]}"
+            return response["d"][0]
+        else:
+            raise ResinResourceNotFound(f"{resource} {params}")
+
+
+class ResinResourceNotFound(exceptions.BalenaException):
+    def __init__(self, message):
+        super(ResinResourceNotFound, self).__init__()
+        self.message = "Resin resource not found {message}".format(message=message)
```

Comment:
I agree, but here the naming was intentionally, I added a comment to explain it to the class: 
"""some API still refer to resin "resources" e.g.:  '/resin/device_type or /resin/release
    use this Exception for request that might fail for these.
    """

Reply:
gotcha

Reply:
I still think that we should ditch the name everywhere that we can. It reduces readability of the code, because not everybody knows that Resin was Balena's old name. Specially on the message: a user reading "Resin resource..." will hardly understand the context of this error.

I would go the extra mile and even suggest that we should wrap all these calls for "resin/" stuff under "balena" naming until they fix their API, but that's my purist side talking, haha.

Reply:
I renamed the ResinResourceException. But the Elephant in the room is of course the name of the tool itself :laughing: 
I had already created a story for collecting improvements here, you can also add more points there  
https://mobilityhouse.atlassian.net/browse/TS-45